### ISSUE TITLE
support for azure workload identity

### ISSUE THREAD

**Issue by ArgTang** [dapr/dapr#5624]
## In what area(s)?
 /area operator
 /area docs

## Describe the feature
Pod identity has officially been [deprecated](https://learn.microsoft.com/en-us/azure/aks/use-azure-ad-pod-identity) and superseeded by [workload identity](https://learn.microsoft.com/en-us/azure/aks/workload-identity-overview). We would like to leverage workload identity together with the dapr install

1. support workload identity in dapr
2. if already supported, create clear docs on how dapr and workload identity work together

related issues:
https://github.com/dapr/docs/issues/2952
https://github.com/dapr/docs/issues/2894

## Release Note

RELEASE NOTE: **ADD** New feature in Dapr.

**Comment by ItalyPaleAle** [comment]
We are tracking this on dapr/components-contrib#1567 - please follow that issue.

TLDR: We are waiting for the upstream SDKs to add support for this. We're hoping it can land in time for Dapr 1.10 but it may be delayed.

**Comment by ArgTang** [comment]
@ItalyPaleAle Thanks, it was not easy to find out what the status was. Ill follow the correct issue

**Comment by berndverst** [comment]
@ArgTang I'm working on this now - targeted for Dapr 1.11 release. The Azure SDK that we depend on finally has this feature and will have a stable release in time for Dapr 1.11 in May. Please follow the issue linked above. Thanks!
